assets/src/dashboard/components/typeahead-input/index.js

```diff
@@ -0,0 +1,215 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { useCallback, useMemo, useRef, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { ReactComponent as SearchIcon } from '../../icons/search.svg';
+import { ReactComponent as CloseIcon } from '../../icons/close.svg';
+import useFocusOut from '../../utils/useFocusOut';
+import { TypeaheadOptions } from '../';
+
+const SearchContainer = styled.div`
+  width: 272px;
+  position: static;
+  display: flex;
+  flex-direction: column;
+  border-radius: ${({ theme, isOpen }) =>
+    isOpen ? theme.border.expandedTypeaheadRadius : 'none'};
+  border: none;
+  box-shadow: ${({ theme, isOpen }) =>
+    isOpen ? theme.boxShadow.expandedTypeahead : 'none'};
+`;
+SearchContainer.propTypes = {
+  isOpen: PropTypes.bool,
+};
+
+const InputContainer = styled.div`
+  position: relative;
+  display: flex;
+  flex-direction: row;
+  width: 100%;
+  height: 48px;
+  padding-top: 15px;
+  padding-bottom: 15px;
+  padding-left: 19px;
+  border-radius: ${({ theme, isOpen }) =>
+    isOpen ? 'none' : theme.border.typeaheadRadius};
+  border: none;
+  border-bottom: ${({ theme, isOpen }) =>
+    isOpen && `1px solid ${theme.colors.gray50}`};
+  color: ${({ theme }) => theme.colors.gray500};
+  background-color: ${({ theme, isOpen }) =>
+    isOpen ? theme.colors.white : theme.colors.gray25};
+`;
+InputContainer.propTypes = {
+  isOpen: PropTypes.bool,
+};
+
+const StyledInput = styled.input`
+  align-self: center;
+  border: none;
+  background-color: transparent;
+  text-overflow: ellipsis;
+  padding: 0 11.95px;
+  margin: auto 0;
+  height: 100%;
+  font-family: ${({ theme }) => theme.fonts.typeaheadInput.family};
+  font-size: ${({ theme }) => theme.fonts.typeaheadInput.size};
+  line-height: ${({ theme }) => theme.fonts.typeaheadInput.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.typeaheadInput.letterSpacing};
+  font-weight: ${({ theme }) => theme.fonts.typeaheadInput.weight};
+  color: ${({ theme }) => theme.colors.gray500};
+
+  &:disabled {
+    cursor: default;
+  }
+`;
+
+const IconContainer = styled.span`
+  margin: auto 0;
+  width: 17.05px;
+  height: 17.05px;
+`;
+
+const ClearInputButton = styled.button`
+  align-self: flex-end;
+  border: none;
+  background-color: transparent;
+  margin: auto 19px auto auto;
+  width: 13.18px;
+  height: 13.18px;
+  padding: 0;
+  color: ${({ theme }) => theme.colors.gray600};
+`;
+
+const TypeaheadInput = ({
+  inputId,
+  items,
+  className,
+  disabled,
+  onChange,
+  maxItemsVisible = 5,
+  placeholder,
+  value = '',
+  ariaLabel,
+  ...rest
+}) => {
+  const [showMenu, setShowMenu] = useState(false);
+  const [inputValue, setInputValue] = useState(value);
+
+  const menuIsOpen = useMemo(() => {
+    return showMenu && items.length > 0 && inputValue;
+  }, [items, showMenu, inputValue]);
```

Comment:
Nit, should the final `&&` statement be `inputValue.length > 0` and that way `menuIsOpen` will always be a boolean.

Also, super nit, should `menuIsOpen` be named `isMenuOpen` to follow the `is<Something>` boolean naming scheme?

Reply:
not super nit, i think this is really important to stick to.